.azure-pipelines/ci-build.yml

```diff
@@ -12,298 +12,252 @@ pr:
-                checkLatest: true
-            - task: DownloadPipelineArtifact@2
-              displayName: Download nupkg from artifacts
-              inputs:
-                artifact: Nugets
-                source: current
-            - task: PowerShell@2
-              displayName: 'Extract release information to pipeline'
-              inputs:
-                targetType: 'filePath'
-                filePath: $(Pipeline.Workspace)\Nugets\scripts\GetNugetPackageVersion.ps1
-                pwsh: true
-                arguments: '-packageDirPath "$(Pipeline.Workspace)/Nugets/"'
-            - task: NuGetCommand@2
-              displayName: 'NuGet push'
-              inputs:
-                command: push
-                packagesToPush: '$(Pipeline.Workspace)/Nugets/Microsoft.Kiota.Abstractions.*.nupkg'
-                nuGetFeedType: external
-                publishFeedCredentials: 'Kiota Nuget Connection'
-            - task: GitHubRelease@1
-              displayName: 'GitHub release (create)'
-              inputs:
-                gitHubConnection: 'Kiota_Release'
-                target: $(Build.SourceVersion)
-                tagSource: userSpecifiedTag
-                tag: 'v$(VERSION_STRING)'
-                title: '$(VERSION_STRING)'
-                releaseNotesSource: inline
-                assets: '!**/**'
-                changeLogType: issueBased
-                isPreRelease : '$(IS_PRE_RELEASE)'
-                addChangeLog : true 
+    - stage: deploy
+      condition: and(contains(variables['build.sourceBranch'], 'refs/heads/main'), succeeded())
+      dependsOn: build
+      jobs:
+        - deployment: deploy_dotnet_abstractions
+          dependsOn: []
+          environment: nuget-org
+          strategy:
+            runOnce:
+              deploy:
+                pool:
+                  vmImage: ubuntu-latest
+                steps:
+                # Install the nuget tool.
+                - task: NuGetToolInstaller@0
+                  displayName: 'Use NuGet >=5.2.0'
+                  inputs:
+                    versionSpec: '>=5.2.0'
+                    checkLatest: true
+                - task: DownloadPipelineArtifact@2
+                  displayName: Download nupkg from artifacts
+                  inputs:
+                    artifact: Nugets
+                    source: current
+                - task: PowerShell@2
+                  displayName: 'Extract release information to pipeline'
+                  inputs:
+                    targetType: 'filePath'
+                    filePath: $(Pipeline.Workspace)\Nugets\scripts\GetNugetPackageVersion.ps1
+                    pwsh: true
+                    arguments: '-packageDirPath "$(Pipeline.Workspace)/Nugets/"'
+                - task: 1ES.PublishNuget@1
+                  displayName: 'NuGet push'
+                  inputs:
+                    packagesToPush: '$(Pipeline.Workspace)/Nugets/Microsoft.Kiota.Abstractions.*.nupkg'
```

Comment:
You might have to set this as well
                        `packageParentPath: '$(Pipeline.Workspace)'`

I know that publish failed for kiota until I did

Reply:
Seeing this too as well at https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/pull/810,
The `DownloadPipelineArtifact@2` seems to be extracting the artifacts in the `$(Pipeline.Workspace)` folder rather than `$(Pipeline.Workspace)\Nugets` folder. So, the paths in the deploy stage may need to be refactored.